framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -4631,6 +4613,442 @@ private void makeGroundTargetType(
+        (AnnotatedDeclaredType) AnnotatedTypeMirror.createType(capturedTypeMirror, this, false);
+
+    captureTypeArgCopier.copy(uncapturedType, capturedType);
+
+    AnnotatedDeclaredType typeDeclaration =
+        (AnnotatedDeclaredType) getAnnotatedType(uncapturedType.getUnderlyingType().asElement());
+
+    // A mapping from type variable to its type argument in the captured type.
+    Map<TypeVariable, AnnotatedTypeMirror> typeVariableMapping = new HashMap<>();
+    // A mapping from the captured type to the annotated captured type.
+    Map<TypeVariable, AnnotatedTypeVariable> capturedTypeMapping = new HashMap<>();
+    List<AnnotatedTypeMirror> newTypeArgs = new ArrayList<>();
+    for (int i = 0; i < typeDeclaration.getTypeArguments().size(); i++) {
+      TypeVariable typeVarTypeMirror =
+          (TypeVariable) typeDeclaration.getTypeArguments().get(i).getUnderlyingType();
+      AnnotatedTypeMirror uncapturedTypeArg = uncapturedType.getTypeArguments().get(i);
+      AnnotatedTypeMirror capturedTypeArg = capturedType.getTypeArguments().get(i);
+      if (uncapturedTypeArg.getKind() == TypeKind.WILDCARD) {
+        if (TypesUtils.isCaptured(capturedTypeArg.getUnderlyingType())) {
+          // The type argument is a captured type. Use the type argument from the newly
+          // created and yet-to-be annotated capturedType. (The annotations are added as
+          // part of capturing the wildcard.)
+          typeVariableMapping.put(typeVarTypeMirror, capturedTypeArg);
+          // Also, add a mapping from the captured type mirror to the annotated captured
+          // type, so that if one captured type refers to another, the correct annotated
+          // type is used.
+          capturedTypeMapping.put(
+              ((AnnotatedTypeVariable) capturedTypeArg).getUnderlyingType(),
+              (AnnotatedTypeVariable) capturedTypeArg);
+          newTypeArgs.add(capturedTypeArg);
+        } else {
+          // If the bounds of the captured wildcard are the same, then it is converted to
+          // a declared type. This seems to be a violation of the JLS, but javac does
+          // this, so the Checker Framework must do it.
+          typeVariableMapping.put(typeVarTypeMirror, capturedTypeArg);
+          newTypeArgs.add(capturedTypeArg);
+          replaceAnnotations(
+              ((AnnotatedWildcardType) uncapturedTypeArg).getSuperBound(), capturedTypeArg);
+        }
+      } else {
+        // The type argument is not a captured type.
+        typeVariableMapping.put(typeVarTypeMirror, uncapturedTypeArg);
+        if (uncapturedTypeArg.getKind() == TypeKind.TYPEVAR) {
+          AnnotatedTypeVariable typeVar = (AnnotatedTypeVariable) uncapturedTypeArg;
+          typeVariableMapping.put(typeVar.getUnderlyingType(), typeVar);
+        }
+        newTypeArgs.add(uncapturedTypeArg);
+      }
+    }
+
+    // Loop through the type arguments and set the annotations of each captured type.
```

Comment:
"each captured type":  This method only captures one type.  Are you talking about the type variables that were constructed to stand in for wildcards -- that is, any type argument that changed as a result of capture conversion?  Maybe we need terminology for them, but "captured type" seems wrong.  This naming choice may be related to my comment about `isCaptured`.

Reply:
I replied to this comme.nt https://github.com/typetools/checker-framework/pull/3922/files#r651288918.